src/services/codefixes/fixImportNonExportedMember.ts

```diff
@@ -0,0 +1,229 @@
+            return undefined;
+        }
+
+        const resolvedModule = getResolvedModule(
+            sourceFile,
+            /*moduleName*/ importDecl.moduleSpecifier.text,
+            /*mode*/ undefined
+        );
+
+        if (!resolvedModule) {
+            return undefined;
+        }
+
+        const originSourceFile = context.program.getSourceFile(
+            resolvedModule.resolvedFileName
+        );
+
+        if (!originSourceFile) {
+            return undefined;
+        }
+
+        return { node, importDecl, originSourceFile };
+    }
+
+    function getNamedExportDeclaration(
+        sourceFile: SourceFile
+    ): ExportDeclaration | undefined {
+        let namedExport;
+
+        for (const statement of sourceFile.statements) {
+            if (
+                isExportDeclaration(statement) &&
+                statement.exportClause &&
+                isNamedExports(statement.exportClause)
+            ) {
+                namedExport = statement;
+            }
+        }
+
+        return namedExport;
+    }
+
+    function compareIdentifiers(s1: Identifier, s2: Identifier) {
+        return compareStringsCaseInsensitive(s1.text, s2.text);
+    }
+
+    function sortSpecifiers(
+        specifiers: ExportSpecifier[]
+    ): readonly ExportSpecifier[] {
+        return stableSort(specifiers, (s1, s2) =>
+            compareIdentifiers(
+                s1.propertyName || s1.name,
+                s2.propertyName || s2.name
+            )
+        );
+    }
+
+    const isVariableDeclarationListWith1Element = (
+        node: Node
+    ): node is VariableDeclarationList =>
+        !(isVariableDeclarationList(node) && node.declarations.length !== 1);
+
+    function doChange(
+        changes: textChanges.ChangeTracker,
+        sourceFile: SourceFile,
+        node: Identifier
+    ): void {
+        const moduleSymbol = sourceFile.localSymbol ?? sourceFile.symbol;
+
+        const localSymbol = moduleSymbol.valueDeclaration?.locals?.get(
+            node.escapedText
+        );
+
+        if (localSymbol === undefined) {
+            return;
+        }
+
+        // Node we need to export is a function
+        if (isFunctionSymbol(localSymbol)) {
+            const node = localSymbol.valueDeclaration;
+
+            if (node === undefined) {
+                return;
+            }
+
+            if (!isDeclarationStatement(node) && !isVariableStatement(node)) {
+                return;
+            }
+
+            return changes.insertExportModifier(sourceFile, node);
+        }
+        // Node we need to export is a variable declaration
+        else if (
+            localSymbol.valueDeclaration &&
+            isVariableDeclarationListWith1Element(
+                localSymbol.valueDeclaration.parent
+            )
+        ) {
+            const node = localSymbol.valueDeclaration.parent;
+
+            return changes.insertExportModifier(sourceFile, node);
+        }

```

Comment:
Do we need special handling for `class` declarations as well?

Reply:
class is captured/handled with `isDeclarationStatement` but I've added some tests and simplified that logic